### PR TITLE
Add simple mode for highlighting QL dbschemes

### PR DESCRIPTION
**Pull request by rneatherway**
@esben-semmle this is a small addition for editing and viewing dbscheme files.

Should I add the `-base` extension to this as well?

### CODE REVIEW

File: misc/emacs/dbscheme-mode.el [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  '("case" "ref" "unique" "of")
+
+  ;; other things to highlight
+  `((,ql/primitive-type-regex . 'font-lock-type-face)
+    ("\\<varchar([0-9]+)" . 'font-lock-type-face)
+    (,ql/at-type-regex 0 'font-lock-type-face)
+    (,ql/predicate-regex 1 'font-lock-variable-name-face))
+
+  ;; auto mode alist
+  '("\\.dbscheme?$")
```
Comment (esben-semmle) [inline misc/emacs/dbscheme-mode.el]: The `?` should not be there. Do we have `foo.dbshcem` files?

File: misc/emacs/dbscheme-mode.el [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  ;; comments
+  '(("//" . nil)
+    ("/*" . "*/"))
+
+  ;; keywords
+  '("case" "ref" "unique" "of")
+
+  ;; other things to highlight
+  `((,ql/primitive-type-regex . 'font-lock-type-face)
```
Comment (esben-semmle) [inline misc/emacs/dbscheme-mode.el]: `ql/primitive-type-regex` is not defined.
We use the double-dash convention for naming internal variables: use `ql--primitive-type-regex` instead.
Ditto below.

File: misc/emacs/dbscheme-mode.el (hunk 1 above)
Comment (rneatherway) [reply misc/emacs/dbscheme-mode.el]: Right, I think I must have copy-pasted from `.qll?` a long time ago!

File: misc/emacs/dbscheme-mode.el (hunk 2 above)
Comment (rneatherway) [reply misc/emacs/dbscheme-mode.el]: I also prefer that convention, updated.

**Comment by rneatherway** [comment]
As the change is minor, I amended my commit.

The `ql--*` variables are undefined unless you require the QL mode first, which the README does recommend.

**Comment by rneatherway** [comment]
Cheers, thanks for the review.

**Pull request merged**